File: android_lifecycle/app/src/main/java/com/example/android_lifecycle/SecondActivity.java
package com.example.android_lifecycle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.LifecycleRegistry;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;

import com.example.android_lifecycle.viewmodels.MyViewModel;

public class SecondActivity extends AppCompatActivity implements LifecycleOwner {

    private LifecycleRegistry mLifecycleRegistry;
    private MyViewModel viewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        mLifecycleRegistry = new LifecycleRegistry(this);
        testViewModel();
    }

    @Override
    public Lifecycle getLifecycle() {
        return mLifecycleRegistry;
    }

    public void testViewModel(){
        viewModel = ViewModelProviders.of(this).get(MyViewModel.class);
        viewModel.init();

    }

}

File: android_hilt/daggersample/src/main/java/top/zcwfeng/daggersample/SecondActivity.kt
package top.zcwfeng.daggersample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import top.zcwfeng.daggersample.dagger.demo2.component.DaggerMyComponent
import top.zcwfeng.daggersample.dagger.demo2.module.DatabaseModule
import top.zcwfeng.daggersample.dagger.demo2.module.HttpModule
import top.zcwfeng.daggersample.data.HttpObject
import javax.inject.Inject

class SecondActivity : AppCompatActivity() {
    @Inject
    lateinit var httpObject:HttpObject
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        // 语法1
//        DaggerMyComponent.create().injectSecondActivity(this)
//        DaggerMyComponent.builder().httpModule(HttpModule())
//            .databaseModule(DatabaseModule())
//            .build()
//            .injectSecondActivity(this)

        (application as DaggerApp).myComponent.injectSecondActivity(this)
        Log.e("httpObject",httpObject.toString() + "--sec")
    }
}